### DISCUSSION TITLE
timestamp field in events_v2 table

### DISCUSSION THREAD

**Discussion by digital88** [plausible/analytics#3682] [Self-Hosted Support] [answered]
Hello.

Is timestamp field in events_v2 table containts date+time in UTC time zone, or the timezone set inside of container where plausible runs?

**Comment by ruslandoga** [comment, marked as answer]
👋 @digital88 

The `events_v2` / `sessions_v2` timestamps are stored as unix timestamps: https://clickhouse.com/docs/en/sql-reference/data-types/datetime

**Reply by digital88** [reply]
Thanks, and how does plausible inserts date into this table: as number or as string?
Docs says there is a (small) difference between both variants of inserting:
```
When inserting datetime as an integer, it is treated as Unix Timestamp (UTC). 1546300800 represents '2019-01-01 00:00:00' UTC. However, as timestamp column has Asia/Istanbul (UTC+3) timezone specified, when outputting as string the value will be shown as '2019-01-01 03:00:00'
When inserting string value as datetime, it is treated as being in column timezone. '2019-01-01 00:00:00' will be treated as being in Asia/Istanbul timezone and saved as 1546290000.
```

**Reply by ruslandoga** [reply]
For inserts Plausible uses RowBinary format and encodes the datetimes as Unix timestamps :)

**Reply by digital88** [reply]
Can you tell please where can I see this in source code?

**Reply by ruslandoga** [reply]
- https://github.com/plausible/analytics/blob/63dec378c2dc35758b7b279c5bc79763ae50ade7/lib/plausible/session/write_buffer.ex#L25-L38
- https://github.com/plausible/analytics/blob/63dec378c2dc35758b7b279c5bc79763ae50ade7/lib/plausible/event/write_buffer.ex#L25-L33
- https://github.com/plausible/analytics/blob/63dec378c2dc35758b7b279c5bc79763ae50ade7/lib/plausible/ingestion/write_buffer.ex#L100
- https://github.com/plausible/ch/blob/8dbda5c6ca1b6b9fd3443c4aab55c3c14e451661/lib/ch/row_binary.ex#L292-L298

**Reply by digital88** [reply]
Thank you!
